lib/src/link_parser.dart

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'charcode.dart';
+import 'text_parser.dart';
+import 'util.dart';
+
+class LinkParser extends TextParser {
```

Comment:
I will implement this `LinkParser` to `LinkSyntax`, this is the reason why I create it separately.